### PR TITLE
fix(snowflake): add output_name for semi-structured access

### PR DESCRIPTION
**Pull request by teraamp**
At least in Snowflake accessing a semi-structured field, i.e. SELECT table.column:field gives an output name of "field", rather than _col_0 or something. This adds this behavior.

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -3961,6 +3961,11 @@ class Between(Predicate):
 class Bracket(Condition):
     arg_types = {"this": True, "expressions": True}
 
+    @property
+    def output_name(self):
+        if self.expressions:
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:3966]: Why was this check necessary? The `expressions` arg is mandatory for `Bracket`.

**Comment by GeorgeSittas** [comment]
It seems like the formatting checks failed; can you please update this PR after first running `make style`?

File: sqlglot/expressions.py (hunk 1 above)
Comment (teraamp) [reply sqlglot/expressions.py]: Fair enough, removed.

**Comment by teraamp** [comment]
> It seems like the formatting checks failed; can you please update this PR after first running `make style`?

Fixed.

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -3961,6 +3961,10 @@ class Between(Predicate):
 class Bracket(Condition):
     arg_types = {"this": True, "expressions": True}
 
+    @property
+    def output_name(self):
+        return self.expressions[0].this
+
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:3967]: On second thought, I'm not sure if this correct.. What if you have a `Bracket` like `a.b[0]`? Or what if you have multiple expressions under the `Bracket`? Just saw that the return type hint was omitted as well; we require `output_name` to return a `str` but `expressions[0].this` isn't necessarily a string value (if we were to go with this approach, perhaps something like `self.expressions[0].name` would be more appropriate).

@tobymao what do you think?

Comment (tobymao) [reply sqlglot/expressions.py:3967]: you can check if this is a literal or a var or something?

Comment (GeorgeSittas) [reply sqlglot/expressions.py:3967]: Yeah, makes sense. An alternative solution could be:

- Check if there's just a single one or more expressions
- If there's only one, output its `name` only if it's a var / string / identifier

Comment (GeorgeSittas) [reply sqlglot/expressions.py:3967]: If we agree on this approach I can take this PR to the finish line.

Comment (tobymao) [reply sqlglot/expressions.py:3967]: just use the output name of the argument

Comment (GeorgeSittas) [reply sqlglot/expressions.py:3967]: https://github.com/tobymao/sqlglot/commit/a81dd14a6de1a50438eae64c2dd20e4841c29572

**Pull request merged**